package lattice;

message Lattice {
  message Node {
    required int32 id = 1;
    optional string label = 2;
    repeated Edge edge = 3;
    extensions 100 to 199;
  }
  
  message Edge {    
    optional int32 id =1;
    optional string label = 2;
    required int32 to_id = 3;
    extensions 100 to 199;
  }

  extensions 1 to 4;
  required int32 start = 5;
  repeated int32 final = 6; 
  repeated Node node = 7;
  extensions 100 to 199;
}

extend Lattice.Node {
  optional bool is_word = 110;
  optional string word = 111; 
  optional int32 original_node = 112; 
  optional bool ignore_node = 113; 
}
